app/controllers/api/api_controller.rb

```diff
@@ -14,7 +14,6 @@ class ApiController < ApplicationController
     rescue_from Api::UnauthorizedTokenError,             with: :not_authenticated
     rescue_from Api::UnsupportedMediaType,               with: :unsupported_media_type
     rescue_from ControlControl::AccessDenied,            with: :not_authorized
-    rescue_from ActiveRecord::StatementInvalid,          with: :bad_query
```

Comment:
The schema is just checking types. The other thing checks if its an
alphabet string or number.
On Jan 10, 2015 8:29 PM, "Campbell Allen" <notifications@github.com> wrote:

> In app/controllers/api/api_controller.rb
> <https://github.com/zooniverse/Panoptes/pull/428#discussion-diff-22763759>
> :
>
> > @@ -14,7 +14,6 @@ class ApiController < ApplicationController
> >      rescue_from Api::UnauthorizedTokenError,             with: :not_authenticated
> >      rescue_from Api::UnsupportedMediaType,               with: :unsupported_media_type
> >      rescue_from ControlControl::AccessDenied,            with: :not_authorized
> > -    rescue_from ActiveRecord::StatementInvalid,          with: :bad_query
>
> I assume Json Schema Validator is checking id formats now so we don't need
> to rescue this one explictly?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zooniverse/Panoptes/pull/428/files#r22763759>.
>